lib/useOnyx.ts

```diff
@@ -46,7 +46,7 @@ type UseOnyxOptions<TKey extends OnyxKey, TReturnValue> = BaseUseOnyxOptions & U
 
 type FetchStatus = 'loading' | 'loaded';
 
-type CachedValue<TKey extends OnyxKey, TValue> = IsEqual<TValue, OnyxValue<TKey>> extends true ? TValue : TKey extends CollectionKeyBase ? NonNullable<OnyxCollection<TValue>> : TValue;
+type CachedValue<TKey extends OnyxKey, TValue> = IsEqual<TValue, OnyxValue<TKey>> extends true ? TValue : TKey extends CollectionKeyBase ? OnyxCollection<TValue> : OnyxEntry<TValue>;
```

Comment:
Have you tested it in E/App? I wonder if it's causing any type errors?

Reply:
I tested, with the changes [only the file mentioned in the GH issue](https://github.com/Expensify/App/issues/44196#issuecomment-2184085451) is failing, which I'm going to fix on E/App!